src/client/datascience/data-viewing/dataViewer.ts

```diff
@@ -61,6 +65,12 @@ export class DataViewer extends WebviewPanelHost<IDataViewerMapping> implements
         );
     }
 
+    private async isSliceDataSupported(dataFrameInfo: IDataFrameInfo) {
+        return (
+            !!dataFrameInfo?.supportsSlicing && (await this.experimentService.inExperiment(Experiments.SliceDataViewer))
```

Comment:
Why do we need an experiment for a backend enhancement? Wouldn't this only affect the UI?

Reply:
Not sure what you mean? The data slicer is a new feature, hence putting it behind a feature flag makes sense. There is a test UI control there now so the feature flag ensures that it's not visible to everyone even if we ship these bits.

Reply:
I don't think we want feature flags in back end code unless necessary. It further complicates the code paths with extra ifs every where. The way you have this coded, the sliceExpression won't be used unless the UI sends it. So I think the feature flag can be only checked in the UI.

That way we test the new code paths with or without the feature flag.

Reply:
I agree.
The UI should just block data slicing features if its not enabled, independent of whether the variable supports slicing.
Also the `dataFrameInfo`  is sent to the UI, hence we can check on the UI whether the variable supports slicing on the UI based on whether:
* Slicing is enabled
* Slicing is supported for the variable.

Reply:
> the sliceExpression won't be used unless the UI sends it.

Yes but the slice control will show up with the variable shape? I'm not sure we'll get the redesigned UI implemented before we cut the February release. I guess I could literally hardcode `supportsSlicing` to false or something in the UI, but I don't think that's what you're suggesting?

Reply:
@rchiodo do you mean query control tower in the UI code?

Reply:
@rchiodo I think I understand your question now. You were asking why we have an experiment for a backend enhancement. The answer is that this PR introduces a temporary UI for testing, and this slice UI (as well as the real one in Part 2) needs to be hidden behind a feature flag for controlled rollout. So here we're checking if the user is in the experiment, and forwarding that information to the UI. I believe this is consistent with other places where we retrieve the experiment info in the backend code and pass it through to the UI via the redux store. Does this help?

FWIW I have dropped the `supportsSlicing` check from this function here and moved it to the UI per Don's feedback that we don't need to pass a derived attribute as a property from the backend to the UI.